### PR TITLE
Feature/switching views

### PR DESCRIPTION
**Pull request by Jayquintana**
### Is this a fix or a feature?
feature
### What is the change?
it adds function to the make the make cover button
### What does it fix?
refactored the change visibility function 
### Where should the reviewer start?
line 23
### Were there any issues that arose?
none, just need to finish line 25
### Refactor checklist 

- Check variable and parameter names: are they descriptive and unique?
 not very descriptive
- Is the code DRY? SRP?
yes
- Is it easy for others to read your code? (explanatory comments needed?)
yes
- Is the syntax consistent? (e.g. use of semicolons, blank lines)
yes except a couple of semicolons 
- Are commit messages formatted correctly?
yes 
### GOALS?
finish line 25 and the rest of iteration 2

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -25,18 +34,49 @@ randomCoverButton.addEventListener("click", randomCover);
 function getRandomIndex(array) {
   return Math.floor(Math.random() * array.length);
 }
-function randomCover() {
+function makeRandomCover() {
```
Comment (BrendaSotel0) [inline src/main.js:37]: We can refractor this

File: src/main.js [hunk 2]
```diff
@@ -25,18 +34,49 @@ randomCoverButton.addEventListener("click", randomCover);
...
   var randomImage = covers[getRandomIndex(covers)];
   var randomTitle = titles[getRandomIndex(titles)];
   var randomTaglineOne = descriptors[getRandomIndex(descriptors)];
   var randomTaglineTwo = descriptors[getRandomIndex(descriptors)];
 
-  var currentCover = new Cover(randomImage, randomTitle, randomTaglineOne, randomTaglineTwo);
+  currentCover = new Cover(randomImage, randomTitle, randomTaglineOne, randomTaglineTwo);
```
Comment (BrendaSotel0) [inline src/main.js:43]: Nice catch!

**Review by BrendaSotel0 — COMMENTED** [review]
Looking good, refractor later.

**Review by BrendaSotel0 — COMMENTED** [review]
Looking good, refractor later.

File: src/main.js [hunk 3]
```diff
@@ -15,8 +21,11 @@ var savedCovers = [
 var currentCover;
 
 // Add your event listeners here 👇
-window.addEventListener("load", randomCover);
-randomCoverButton.addEventListener("click", randomCover);
+window.addEventListener("load", makeRandomCover);
+randomCoverButton.addEventListener("click", makeRandomCover);
+makeCoverButton.addEventListener("click", showForm);
+homeButton.addEventListener("click", loadHome);
+viewSavedCoversButton.addEventListener("click", showSavedCovers);
```
Comment (itsnameissteven) [inline src/main.js:28]: Nice descriptive names. looks good

File: src/main.js [hunk 4]
```diff
@@ -25,18 +34,49 @@ randomCoverButton.addEventListener("click", randomCover);
 function getRandomIndex(array) {
   return Math.floor(Math.random() * array.length);
 }
-function randomCover() {
+function makeRandomCover() {
   var randomImage = covers[getRandomIndex(covers)];
   var randomTitle = titles[getRandomIndex(titles)];
   var randomTaglineOne = descriptors[getRandomIndex(descriptors)];
   var randomTaglineTwo = descriptors[getRandomIndex(descriptors)];
```
Comment (itsnameissteven) [inline src/main.js:42]: This is nice and readable. I like it, If you want to challenge yourself you could try and write a function above this one that can take in an argument and returns what you are looking for. If you do write that function you can take out the above four lines and add the function to line 43

File: src/main.js [hunk 5]
```diff
@@ -25,18 +34,49 @@ randomCoverButton.addEventListener("click", randomCover);
...
+
+function hideVariable(variable){
+  variable.classList.add("hidden")
+
+}
+
+function showVariable(variable){
+  variable.classList.remove("hidden")
+}
+
```
Comment (itsnameissteven) [inline src/main.js:59]: nice reusable functions.

File: src/main.js [hunk 6]
```diff
@@ -25,18 +34,49 @@ randomCoverButton.addEventListener("click", randomCover);
...
+  hideVariable(formPage);
+}
 
-  console.log(currentCover);
+function showSavedCovers() {
+  hideVariable(mainPage);
+  showVariable(savedCoversPage);
+  hideVariable(randomCoverButton);
+  hideVariable(saveCoverButton);
+  showVariable(homeButton);
```
Comment (itsnameissteven) [inline src/main.js:81]: These are clean and readable but you are writing out the function a bunch of times. I think it is good the way it is but I wonder if you could combine the hide and show functions some how. maybe the function hides everything first and then pass the elements you want to show as an argument? Keep an eye on these as the project grows and you might see a pattern

**Review by itsnameissteven — APPROVED** [review]
Looks good. I don't think you need to change anything at this point but I commented on a few places if you feel like challenging yourself.

File: src/main.js (hunk 6 above)
Comment (itsnameissteven) [reply src/main.js:81]: If you go down that path don't sacrifice readability for it. Readability > complex code

**Pull request merged**